database-plugins/src/main/java/co/cask/hydrator/plugin/db/batch/source/DBSource.java

```diff
@@ -158,39 +104,26 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
    * we get the message and throw it as BadRequestException
    */
   @Path("getSchema")
-  public Map<String, String> getSchema(GetSchemaRequest request,
+  public Schema getSchema(GetSchemaRequest request,
```

Comment:
not sure why this is in outdated diff.. 

we already have SchemaTypeAdapter registered for GSON in CDAP.
`private static final Gson GSON = new GsonBuilder()
    .registerTypeAdapter(Schema.class, new SchemaTypeAdapter())
    .registerTypeAdapter(PluginClass.class, new PluginClassDeserializer())
    .create();` in ArtifactHttpHandler